### PR TITLE
Move shopper-side Checkout E2E tests to Playwright

### LINKED ISSUE #11895: E2E tests migration: shopper -> checkout
Migrate E2E tests from `tests/e2e-jest/specs/shopper/cart-checkout/checkout.test.js` to Playwright.

### PR DESCRIPTION
**Pull request by tarhi-saad**
## What

Moves shopper-side Checkout E2E tests to Playwright - we skipped them in Puppeteer to get old E2Es working again.

Removes the old Puppeteer shopper checkout tests.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11895

## Why

To 1) re-enable a skipped test and 2) move from Puppeteer to Playwright

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Check playwright E2E tests pass, both `block_theme` and `classic_theme`

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

N/A

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> N/A

### CODE REVIEW

**Review by wavvves — APPROVED** [review]

**Review by nielslange — COMMENTED** [review]
Thanks for migrating these tests, @tarhi-saad. However, I noticed that the line breaks between different sections seem to be missing in the migrated test file ([tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts](https://github.com/woocommerce/woocommerce-blocks/blob/026bc530db8a3509f9d92920c61d9434c30f93b0/tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts), lines 109-315). This file currently has only 2 line breaks in the 206 lines, impacting its readability. Could you share the reasoning behind this approach? Adding appropriate line breaks can enhance the clarity and maintainability of the code.

**Comment by tarhi-saad** [comment]
I added some line breaks as suggested, @nielslange! 🙌

File: tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts [hunk 1]
```diff
@@ -88,3 +105,225 @@ test.describe( 'Shopper → Account', () => {
...
+			'thisShouldRemainHere@mail.com'
+		);
+		await page
+			.getByLabel( 'Email address' )
+			.fill( 'thisShouldRemainHereToo@mail.com' );
+		await page.getByLabel( 'First name' ).fill( 'Test FirstName' );
+		await expect( page.getByLabel( 'Email address' ) ).toHaveValue(
+			'thisShouldRemainHereToo@mail.com'
+		);
+	} );
```
Comment (nielslange) [inline tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts]:
```suggestion
	} ) => {
		await frontendUtils.emptyCart();
		await frontendUtils.goToShop();
		await frontendUtils.addToCart( SIMPLE_PHYSICAL_PRODUCT_NAME );
		await frontendUtils.goToCheckout();
		
		await page.getByRole( 'radio', { name: 'Local Pickup free' } ).click();
		await expect( page.getByLabel( 'Testing' ).last() ).toBeVisible();
		
		await checkoutPageObject.fillInCheckoutWithTestData();
		
		await page
			.getByLabel( 'Email address' )
			.fill( 'thisShouldRemainHere@mail.com' );
		await page.getByRole( 'radio', { name: 'Shipping from free' } ).click();
		await expect(
			page.getByRole( 'radio', { name: 'Flat rate shipping' } ).first()
		).toBeVisible();
		await expect( page.getByLabel( 'Email address' ) ).toHaveValue(
			'thisShouldRemainHere@mail.com'
		);
		
		await page
			.getByLabel( 'Email address' )
			.fill( 'thisShouldRemainHereToo@mail.com' );
		await page.getByLabel( 'First name' ).fill( 'Test FirstName' );
		await expect( page.getByLabel( 'Email address' ) ).toHaveValue(
			'thisShouldRemainHereToo@mail.com'
		);
	} );
```

**Review by nielslange — COMMENTED** [review]
Thanks for applying the changes, @tarhi-saad. I left one more suggestion. 😀

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @tarhi-saad. The changes look great. Let's ⛴️ this! 🙌
